File: examples/nodejs/batch-conversion.js
/**
 * Batch Image Conversion Example
 *
 * This example shows how to convert multiple images in batch
 */

const ConvertorioClient = require('../../libs/nodejs');
const fs = require('fs');
const path = require('path');

const client = new ConvertorioClient({
    apiKey: 'your_api_key_here'
});

async function convertBatch(files, targetFormat) {
    console.log(`Converting ${files.length} files to ${targetFormat.toUpperCase()}...\n`);

    const results = [];

    for (let i = 0; i < files.length; i++) {
        const file = files[i];
        console.log(`[${i + 1}/${files.length}] Converting ${path.basename(file)}...`);

        try {
            const result = await client.convertFile({
                inputPath: file,
                targetFormat
            });

            results.push({
                success: true,
                file,
                outputPath: result.outputPath
            });

            console.log(`✓ Completed: ${result.outputPath}\n`);

        } catch (error) {
            results.push({
                success: false,
                file,
                error: error.message
            });

            console.error(`✗ Failed: ${error.message}\n`);
        }
    }

    return results;
}

async function main() {
    // Example: Convert all PNG files in a directory to JPG
    const inputDir = './images';
    const targetFormat = 'jpg';

    // Get all PNG files
    const files = fs.readdirSync(inputDir)
        .filter(file => file.toLowerCase().endsWith('.png'))
        .map(file => path.join(inputDir, file));

    if (files.length === 0) {
        console.log('No PNG files found in', inputDir);
        return;
    }

    const results = await convertBatch(files, targetFormat);

    // Summary
    const successful = results.filter(r => r.success).length;
    const failed = results.filter(r => !r.success).length;

    console.log('\n========================================');
    console.log('Batch Conversion Summary');
    console.log('========================================');
    console.log(`Total: ${results.length}`);
    console.log(`✓ Successful: ${successful}`);
    console.log(`✗ Failed: ${failed}`);
    console.log('========================================\n');

    if (failed > 0) {
        console.log('Failed conversions:');
        results.filter(r => !r.success).forEach(r => {
            console.log(`  - ${r.file}: ${r.error}`);
        });
    }
}

main();

File: examples/ruby/batch_conversion.rb
#!/usr/bin/env ruby

# Batch Conversion Example
#
# This example shows how to convert multiple files in a directory

require_relative '../../libs/ruby/lib/convertorio'

client = Convertorio::Client.new(
  api_key: 'your_api_key_here'
)

# Get all PNG files from a directory
input_dir = './images'
files = Dir.glob("#{input_dir}/*.png")

if files.empty?
  puts "No PNG files found in #{input_dir}"
  exit(0)
end

puts "Found #{files.size} PNG files to convert\n\n"

# Convert all files to WebP
converted = 0
failed = 0

files.each_with_index do |file, index|
  begin
    puts "[#{index + 1}/#{files.size}] Converting: #{File.basename(file)}"

    result = client.convert_file(
      input_path: file,
      target_format: 'webp'
    )

    size_kb = (result[:file_size] / 1024.0).round(2)
    puts "✓ Converted: #{File.basename(result[:output_path])} (#{size_kb} KB)\n\n"
    converted += 1

  rescue Convertorio::Error => e
    puts "✗ Failed: #{e.message}\n\n"
    failed += 1
  end
end

puts "\n" + "="*50
puts "Batch conversion completed!"
puts "Successful: #{converted}"
puts "Failed: #{failed}"
puts "Total: #{files.size}"
